src/main/java/org/gaul/s3proxy/junit/S3ProxyExtension.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2014-2020 Andrew Gaul <andrew@gaul.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.s3proxy.junit;
+
+import java.net.URI;
+
+import org.gaul.s3proxy.AuthenticationType;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/**
+ * A JUnit 5 Extension that manages an S3Proxy instance which tests
+ * can use as an S3 API endpoint.
+ */
+public final class S3ProxyExtension
+        implements AfterAllCallback, BeforeAllCallback {
```

Comment:
The JUnit4 rule extends `ExternalResource`  and then overrides `before()` which is the equivalent of `beforeEach` in JUnit5 not `beforeAll`. So to be consistent you need to change this to implement `BeforeEachCallback`. The same goes for `AfterEachCallback`.